core/lib/common/worker.js

```diff
@@ -326,17 +326,39 @@ module.exports = class Worker {
 	}
 
 	async executeCommandInWorkerHost(command) {
-		let config = {};
-		const result = await utils.executeCommandOverSSH(
-			`${this.sshPrefix}${command}`,
+		let config = {
+			host: this.workerHost,
+			port: this.workerPort,
+			username: this.workerUser,
+		}
+
+		return retry(
+			async () => {
+				let result = {}
+				try {
+					result = await utils.executeCommandOverSSH(`${this.sshPrefix}${command}`, config);
+				} catch (err){
+					console.error(err)
```

Comment:
Is this going to be more verbose than we need? How does it look in your tests?

Reply:
@klutchell this change is actually to the `executeCommandInWorkerHost` function, not `executeCommandInHostOs` - this brings it in line with the existing  `executeCommandInHostOs`.

Reply:
The error logs will look like they do in `executeCommandInHostOS` :

```
client_1  | [2022-06-14T14:24:46.273Z][1c1f697-os] Error: No response from server
client_1  |     at Socket.<anonymous> (/usr/app/node_modules/ssh2/lib/client.js:764:19)
client_1  |     at Socket.emit (events.js:400:28)
client_1  |     at TCP.<anonymous> (net.js:686:12)
client_1  |     at TCP.callbackTrampoline (internal/async_hooks.js:130:17)
client_1  | [2022-06-14T14:24:46.273Z][1c1f697-os] Error while performing SSH command "[[ ! -f /tmp/reboot-check ]] && echo pass", will retry...
```

Very verbose not going to lie, but imo while we stabilise things I wanted as much detail as possible

Reply:
Can we change these to just capture `err.message` if it exists, and avoid the stack trace?

I agree with verbosity but the stack hasn't been much help has it?

Reply:
agreed, done